### PR TITLE
Adding a subscription for translations content

### LINKED ISSUE #19945: Subscribe to receive translations from content updater.

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #19945 |

## Description

This PR creates a new registration for the translations topic and subscribes to this content to update the corresponding DB.

## Tests

A manual test with the testtool was performed.
The following file was used 

<details><summary>Details</summary>
<p>

```
root@ubuntu-jammy:/tmp/vd_updater/contents# cat file.json 
{
    "data": [
        {
            "type": "create",
            "resource": "WT-3",
            "payload": {
                "target": "windows",
                "source": {
                    "vendor": "^The Apache Software",
                    "product": "^Apache Tomcat.*",
                    "version": ""
                },
                "translation": [
                    {
                        "vendor": "apache",
                        "product": "tomcat",
                        "version": ""
                    }
                ],
                "action": [
                    0,
                    1
                ]
            }
        },
        {
            "type": "create",
            "resource": "WT-4",
            "payload": {
                "target": "windows2",
                "source": {
                    "vendor": "^The Apache Software2",
                    "product": "^Apache Tomcat2.*",
                    "version": ""
                },
                "translation": [
                    {
                        "vendor": "apache2",
                        "product": "tomcat2",
                        "version": ""
                    }
                ],
                "action": [
                    0,
                    1
                ]
            }
        }
    ]
}
```

</p>
</details> 

The message was processed successfully 

And the entries are saved in the DB

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp [hunk 1]
```diff
@@ -186,6 +187,31 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
                     // TO DO -> call to reinitialize the database.
                 }
             });
+
+        // Subscription to vulnerability detector translations update events.
+        m_translationsSubscription = std::make_unique<TRouterSubscriber>(
+            updaterPolicy.at("translationsTopic"), "vulnerability_feed_manager", isLocalSubscriber);
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:193]: The router topic name could be in a constant. So subscribers and publishers could use the same constant to eliminate the possibility of being different.

Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:193]: The router's subscriberId argument could be in a constant. So different subscribers in this class could use the same constant.

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 2]
```diff
@@ -124,6 +124,7 @@ class PolicyManager final : public Singleton<PolicyManager>
                 Utils::parseStrToTime(newPolicy.at("vulnerability-detection").at("feed-update-interval"));
             newPolicy["updater"]["ondemand"] = false;
             newPolicy["updater"]["topicName"] = "vulnerability_feed_manager";
+            newPolicy["updater"]["translationsTopic"] = "vulnerability_feed_manager_translations";
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:127]: The router's subscriberId argument could be in a constant. So different subscribers in this class could use the same constant.
In this case the translationTopic don't match with the one used in databaseFeedManager.hpp

**Review by matias-braida — COMMENTED** [review]
GJ

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:127]: Here we are setting the default configuration, and assigning `vulnerability_feed_manager_translations` as a `topicName`.

In ` databaseFeedManager.hpp` we are using this same configuration value `["updater"]["translationsTopic"]`.
The `topicName` and the `subscriberId` aren't required to be the same.

Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:127]: You are right. My bad

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp]: A constant was added.

**Review by matias-braida — APPROVED** [review]
LGTM

**Pull request closed without merging**